### PR TITLE
KEYCLOAK-3000: fixed the saml brokering tests - 1.9.x

### PR DESCRIPTION
**Pull request by alicerum**

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java [hunk 1]
```diff
@@ -157,23 +174,33 @@ public void logInAsUserInIDP() {
...
+        }
+
         driver.navigate().to(getAuthRoot()
                 + "/auth/realms/" + providerRealmName()
                 + "/protocol/" + "openid-connect"
-                + "/logout");
+                + "/logout?redirect_uri=" + encodedAccount);
 
-        driver.navigate().to(getAuthRoot() + "/auth/realms/" + providerRealmName() + "/account");
+        waitForPage("log in to " + providerRealmName());
```
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java]: This shouldn't be necessary as the page is not an ajax page. Once driver.navigate is complete the page should be loaded.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java [hunk 2]
```diff
@@ -113,22 +115,37 @@ protected IdentityProviderRepresentation createIdentityProvider(String alias, St
         return identityProviderRepresentation;
     }
 
+    private void waitForPage(String title) {
+        long startAt = System.currentTimeMillis();
+
+        while (!driver.getTitle().toLowerCase().contains(title)
+                && System.currentTimeMillis() - startAt < 200) {
+            try {
+                Thread.sleep(5);
```
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java]: Do NOT add 5 second sleeps. Any waiting for elements has to pull periodically and not just sleep for a long time (5 seconds * 1000 tests is a long time!).

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java (hunk 1 above)
Comment (wyvie) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java]: But it isn't. Often with SAML after navigate we are on "SAML HTTP Post Binding" page. Which has onload body script "<BODY Onload=\"document.forms[0].submit()\">"
Oidc always works correctly.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java (hunk 2 above)
Comment (wyvie) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java]: Thread.sleep() accepts milliseconds, not seconds

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java (hunk 1 above)
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java]: You're right. Didn't think about SAML.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java (hunk 2 above)
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBrokerTest.java]: Yup. I'm to used to our own seconds based time.

**Pull request merged**